### PR TITLE
chore: cleanup packages and names

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary
Our package names for all components of Firefly are inconsistent and could benefit from some common naming. It's best to use `firefly-*` as a prefix. Additionally, we will be adding the extension actor to the backend soon, making the name `wallet-actor-*` not technically correct.

:warning: `shared` __will NOT__ be renamed here as this package is about to be refactored.

### Changelog
```
- Change "wallet-actor-system" to "firefly-actor-system"
- Change "wallet-actor-system-capacitor-binding" to "firefly-actor-system-capacitor-bindings"
- Change "wallet-nodejs-binding" to "firefly-actor-system-nodejs-bindings"
- Change "mobile-wallet" to "firefly-mobile"
- Change "desktop" to "firefly-desktop"
```

## Relevant Issues
#2385 

## Type of Change
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [x] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

### Platforms
- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
_N/a_

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-    "name": "desktop",
-    "productName": "Firefly",
+    "name": "firefly-desktop",
+    "productName": "Firefly Desktop",
```
Comment (rajivshah3) [inline packages/desktop/package.json:3]: I believe this affects the naming of the app on desktop as well (therefore potentially impacting file paths), so we should do this change in a separate PR too. Testing this change will involve testing the entire update process to ensure app data is not lost when we change this, as well as ensuring CI and deployment scripts don't break

File: packages/mobile/package.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
-    "name": "mobile-wallet",
+    "name": "firefly-mobile",
+    "productName": "Firefly Mobile",
```
Comment (rajivshah3) [inline packages/mobile/package.json:3]: I think `productName` is specific to `electron-builder`, so this might not make a difference for Firefly Mobile

**Review by rajivshah3 — CHANGES_REQUESTED** [review]

File: packages/desktop/package.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
-    "name": "desktop",
-    "productName": "Firefly",
+    "name": "firefly-desktop",
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/desktop/package.json:2]: No problem! We can make a task for this and keep track of it there :+1:

File: packages/mobile/package.json (hunk 2 above)
Comment (maxwellmattryan) [reply packages/mobile/package.json:3]: Okay, I'll revert it just in case.

**Review by rajivshah3 — APPROVED** [review]
Builds properly on desktop and mobile, so I'll merge so that CI and local builds are fixed on `develop`

**Pull request merged**